sdk/provisioning/Azure.Provisioning.CloudMachine/src/CDKLevel3/CloudMachineCommands.cs

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System;
+using System.ClientModel.TypeSpec;
+using System.ClientModel.Primitives;
+using Azure.AI.OpenAI;
+using Azure.Core;
+using Azure.Identity;
+using System.Text.Json;
+
+namespace Azure.CloudMachine;
+
+public class CloudMachineCommands
+{
+    public static bool Execute(string[] args, Action<CloudMachineInfrastructure>? configure = default, bool exitProcessIfHandled = true)
+    {
+        if (args.Length < 1) return false;
+
+        string cmid = AzdHelpers.ReadOrCreateCmid();
+        CloudMachineInfrastructure cmi = new(cmid);
+        if (configure != default)
+        {
+            configure(cmi);
+        }
+
+        if (args[0] == "-bicep")
+        {
+            GenerateBicep(cmi);
+            return Handled(exitProcessIfHandled);
+        }
+
+        if (args[0] == "-tsp")
+        {
+            GenerateTsp(cmi);
+            return Handled(exitProcessIfHandled);
+        }
+
+        if (args[0] == "-ai")
+        {
+            string? option = (args.Length > 1) ? args[1] : default;
+            ListAzureOpenaAIModels(cmid, option);
+            return Handled(exitProcessIfHandled);
+        }
+
+        return false;
+
+        bool Handled(bool exitProcessIfHandled)
+        {
+            if (exitProcessIfHandled) Environment.Exit(0);
+            return true;
+        }
+    }
+
+    // this implements: https://learn.microsoft.com/en-us/rest/api/azureopenai/models/list
+    private static void ListAzureOpenaAIModels(string cmid, string? option)
```

Comment:
This seems like a method that should be on [CloudMachineClient.AzureOpenAIExtensions](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cloudmachine/Azure.CloudMachine/src/extensions/AzureOpenAIExtensions.cs). The other commands are all local while this one operates on live azure resources.

Reply:
moving it also eliminates the need for all the extra dependencies like Identity, the RestClient, and the bespoke auth policy.

Reply:
The main scenario for this is during provisioning, not at CM app runtime. I want to allow model name and version to be optional and then grab the best models (according to some heuristics) when the bicep is generated. But I ran into problems that I need to talk to some people about and so for now I just turned the code into a command